application/apps/indexer/parsers/src/dlt/fmt.rs

```diff
@@ -281,6 +285,17 @@ impl<'a> Serialize for FormattableMessage<'a> {
                     None => state.serialize_field("payload", "[Unknown CtrlCommand]")?,
                 }
             }
+            PayloadContent::NetworkTrace(slices) => {
+                state.serialize_field("app-id", &ext_header_app_id)?;
+                state.serialize_field("context-id", &ext_header_context_id)?;
+                state.serialize_field("message-type", &ext_header_msg_type)?;
+                let arg_string = slices
+                    .iter()
+                    .map(|slice| format!("{:02X?}", slice))
+                    .collect::<Vec<String>>()
+                    .join("|");
```

Comment:
I'm not sure if using `reduce()` here could solve this better:
```rust
                let arg_string = slices
                    .iter()
                    .map(|slice| format!("{:02X?}", slice))
                    .reduce(|mut acc, s| {
                        acc.push('|');
                        acc.push_str(&s);
                        acc
                    })
                    .unwrap_or_default();
```
If you think this is better, then we can apply it to `PayloadContent::Verbose` match arm on line 240

Reply:
I don't see how is reduce expressing the intent better here.

Reply:
The point of reduce here is to avoid allocation for the strings into a vector and then apply join on it, by starting with one string and appending each string to it. 
I think it wouldn't have a big impact on the performance though and we can leave like this to favor code readability here.

Reply:
Using 'join' on string concatenation seems indeed to be the way of choice within the code-base at multiple occasions, while 'reduce' seems currently only to be used within some arithmetic operations. Therefore I would prefer to stay with the present solution, yet recommend a separe PR for a global change hereon, if and only if a performance impact evaluation would show some measurable benefit for the tradeoff of code readability.